android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -1245,13 +1245,13 @@ Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv *
   jclass const routeAltitudeLimitsClass = env->GetObjectClass(routeAltitudeLimits);
   ASSERT(routeAltitudeLimitsClass, ());
 
-  static jfieldID const minRouteAltitudeField = env->GetFieldID(routeAltitudeLimitsClass, "minRouteAltitude", "I");
-  ASSERT(minRouteAltitudeField, ());
-  env->SetIntField(routeAltitudeLimits, minRouteAltitudeField, minRouteAltitude);
+  static jfieldID const totalAscentField = env->GetFieldID(routeAltitudeLimitsClass, "totalAscent", "I");
+  ASSERT(totalAscentField, ());
+  env->SetIntField(routeAltitudeLimits, totalAscentField, totalAscent);
 
-  static jfieldID const maxRouteAltitudeField = env->GetFieldID(routeAltitudeLimitsClass, "maxRouteAltitude", "I");
-  ASSERT(maxRouteAltitudeField, ());
-  env->SetIntField(routeAltitudeLimits, maxRouteAltitudeField, maxRouteAltitude);
+  static jfieldID const totalDescentField = env->GetFieldID(routeAltitudeLimitsClass, "totalDescent", "I");
+  ASSERT(totalDescentField, ());
+  env->SetIntField(routeAltitudeLimits, totalDescentField, totalDescent);
```

Comment:
Is there a warning here about uint to int conversion? static_cast<int32_t> is needed, probably also asserted that the uint value is not greater than an int one.

Reply:
Added explicit cast. But without aeesrtion